lang/LangPrimSource/PyrFilePrim.cpp

```diff
@@ -1326,6 +1326,74 @@ int prPipeOpen(struct VMGlobals *g, int numArgsPushed)
 	return errNone;
 }
 
+int prPipeOpenArgv(struct VMGlobals *g, int numArgsPushed)
+{
+	PyrSlot *a, *b, *c;
+	char mode[12];
+
+	a = g->sp - 2;
+	b = g->sp - 1; //argv
+	c = g->sp;     //mode
+
+	if (NotObj(b)) return errWrongType;
+
+	if (NotObj(c) || !isKindOf(slotRawObject(c), class_string))
+		return errWrongType;
+	if (slotRawObject(c)->size > 11) return errFailed;
+
+	PyrFile *pfile = (PyrFile*)slotRawObject(a);
+
+	PyrObject *obj = slotRawObject(b);
```

Comment:
could use a more descriptive name than `obj` here, maybe `argvColl` since it's supposed to be a collection containing `argv`.

Reply:
ok